iree/samples/simple_embedding/device_dylib.c

```diff
@@ -12,23 +12,24 @@
 #include "iree/hal/api.h"
 #include "iree/hal/local/executable_loader.h"
 #include "iree/hal/local/loaders/embedded_library_loader.h"
-#include "iree/hal/local/loaders/legacy_library_loader.h"
 #include "iree/hal/local/task_device.h"
 #include "iree/task/api.h"
 
+// Compiled module embedded here to avoid file IO:
+#include "iree/samples/simple_embedding/simple_embedding_test_bytecode_module_dylib_riscv_64_c.h"
+#include "iree/samples/simple_embedding/simple_embedding_test_bytecode_module_dylib_x86_64_c.h"
+
 iree_status_t create_sample_device(iree_hal_device_t** device) {
   // Set paramters for the device created in the next step.
   iree_hal_task_device_params_t params;
   iree_hal_task_device_params_initialize(&params);
 
-  iree_hal_executable_loader_t* loaders[2] = {NULL, NULL};
+  iree_hal_executable_loader_t* loader = NULL;
   iree_host_size_t loader_count = 0;
   IREE_RETURN_IF_ERROR(iree_hal_embedded_library_loader_create(
       iree_hal_executable_import_provider_null(), iree_allocator_system(),
-      &loaders[loader_count++]));
-  IREE_RETURN_IF_ERROR(iree_hal_legacy_library_loader_create(
-      iree_hal_executable_import_provider_null(), iree_allocator_system(),
-      &loaders[loader_count++]));
+      &loader));
+  loader_count++;
```

Comment:
can drop loader_count (as it's always 1)

Reply:
Done.